From da9aae4807ef14dea9cea029180cb568535b7edb Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@freescale.com>
Date: Fri, 18 May 2012 13:54:30 +0300
Subject: [PATCH 21/27] gianfar: Replace WARN_ON with warning message

Replace WARN_ON to avoid stack dump, when the H/W TCP receive
option (CONFIG_GFAR_HW_TCP_RECEIVE_OFFLOAD) has been selected for
net device instances that don't support it (i.e. TSEC, eTSEC1.x,
or insufficient number h/w Rx queues (set via dts)).
In such cases, the feature is being disabled at runtime, but may
incur performance penalties.

Signed-off-by: Claudiu Manoil <claudiu.manoil@freescale.com>
[Kevin: Original patch taken from FSL
QorIQ-SDK-V1.2-SOURCE-20120614-yocto.iso image]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/freescale/gianfar.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/gianfar.c b/drivers/net/ethernet/freescale/gianfar.c
index a26f4c9..53a35da 100644
--- a/drivers/net/ethernet/freescale/gianfar.c
+++ b/drivers/net/ethernet/freescale/gianfar.c
@@ -1492,7 +1492,9 @@ static int gfar_probe(struct platform_device *ofdev)
 		priv->hw_tcp.en = 0;
 	/* not a good idea to activate this feature if this gfar instance
 	 * does not support it */
-	WARN_ON(!priv->hw_tcp.en);
+	if (!priv->hw_tcp.en)
+		netdev_warn(dev,
+			"H/W TCP receive offload not supported (disabled)!");
 #endif
 
 	/* Initialize the filer table */
-- 
1.7.9.7

